hooks/node_handlers/base_render_handler.py

```diff
@@ -0,0 +1,369 @@
+    """
+    Base class for all render output nodes.
+    """
+
+    SGTK_PASS_NAME = "sgtk_pass_name"
+
+    # aovs
+    AOV_COUNT = ""
+    AOV_NAME_TMPL = "{}"
+    AOV_FILE_TMPL = "{}"
+    AOV_USE_FILE_TMPL = "{}"
+
+    # archives
+    ARCHIVE_ENABLED = ""
+    ARCHIVE_OUTPUT = ""
+
+    # templates
+    AOV_WORK_TEMPLATE = "aov_work_template"
+    AOV_PUBLISH_TEMPLATE = "aov_publish_template"
+    ARCHIVE_WORK_TEMPLATE = ""
+    ARCHIVE_PUBLISH_TEMPLATE = ""
+
+    # strings
+    AOV_ERROR = ""
+    
+    def _get_template_fields_from(self, node, additional_fields={}):
+        """
+        Get shotgun template fields from the given node and update with any additional
+        fields.
+
+        :param node: A :class:`hou.Node` instance.
+        :param dict additional_fields: Any fields to override with.
+
+        :returns: The template fields updated with the additional_fields
+        """
+        output_parm = node.parm(self.OUTPUT_PARM)
+        file_path = output_parm.unexpandedString()
+        fields = self.get_work_template(node).validate_and_get_fields(file_path)
+        if not fields:
+            raise sgtk.TankError("Can not extract fields from file path")
+        if "SEQ" in fields:
+            fields["SEQ"] = "FORMAT: $F"
+        fields.update(additional_fields)
+        return fields
+    
+    def generate_aov_path(self, node, channel, template_name):
+        """
+        Generate the file path for the given aov name.
+
+        :param node: A :class:`hou.Node` instance.
+        :param str channel: The aov name.
+        :param str template_name: The name of the shotgun template to use.
+
+        :returns: The file path for the aov output.
+        """
+        try:
+            fields = self._get_template_fields_from(node)
+            template = self._get_template(template_name)
+        except sgtk.TankError as error:
+            return str(error)
+        if channel:
+            fields["channel"] = channel
+        try:
+            file_path = template.apply_fields(fields)
+        except sgtk.TankError:
+            return self.AOV_ERROR
+        return file_path
+
+    def _lock_parms(self, node, lock):
+        """
+        Lock parms on the node is shotgun is enabled.
+
+        :param node: A :class:`hou.Node` instance.
+        :param bool lock: Lock parms if True.
+        """
+        self._lock_aov_parms(node, lock)
+        archive_output = node.parm(self.ARCHIVE_OUTPUT)
+        archive_output.lock(lock)
+
+    def _update_aov_paths(self, node):
+        """
+        Update all the aov output paths on the node.
+
+        :param node: A :class:`hou.Node` instance.
+        """
+        aov_count = node.parm(self.AOV_COUNT)
+        count = aov_count.eval() + 1
+        for index in range(1, count):
+            aov_name = node.parm(self.AOV_NAME_TMPL.format(index))
+            sgtk_aov_name = node.parm(
+                self.SGTK_AOV_NAME_TMPL.format(index)
+            )
```

Comment:
Fit into 1 line < 79 chars

```suggestion
            sgtk_aov_name = node.parm(self.SGTK_AOV_NAME_TMPL.format(index))
```